### ISSUE TITLE
Production mode

### ISSUE THREAD

**Issue by 7flash** [webui-dev/deno-webui#25]
Since this commit packages are not auto downloaded https://github.com/webui-dev/deno-webui/commit/cdac5cdfb29c4e8bb59596e8ac814eae80408edb

Therefore compiled binary files with error

Error: Could not open library: Could not open library: dlopen

**Comment by AlbertShown** [comment]
I guess you need to run `bootstrap.sh` or `bootstrap.bat`.

**Comment by 7flash** [comment]
When I run bootstrap.sh as developer, it will download library compiled for my specific OS; but what about users? when I distribute my deno app as compiled executable, it have had correct behaviour of downloading library for user OS at the time of first run, but with recent commit it fails to find it.

**Comment by AlbertShown** [comment]
That's true. In this case, https://deno.land/x/webui should be updated to solve this issue.
@hassandraga.

**Comment by ttytm** [comment]
The library is only needed during the compilation step. Distributing would usually happen via a CI and users could download the distributable of the app.

The lib would only be needed if a users would compile the app from source, in that case the webui bootstrap step should be part of the build instructions.

**Comment by AlbertShown** [comment]
> the webui bootstrap step should be part of the build instructions

Yes @ttytm, The developers should use the bootstrap step 👍 and the final distributed app depends on how it is packaged, for example:

1. The developer simply compresses his project and shares, then the dynamic libs are already in the project (bootstrap step)
2. The developer shares his project as a GitHub repo, then he should add the bootstrap step in his readme preparation steps
3. The developer has a single executable `deno compile ...` that imports multiple Deno packages in runtime `import { xxx } from "https://deno.land/x/xxx"`... then https://deno.land/x/webui should have already the binaries "_This capability is one of the features in Deno_"... Or deno-webui can easily run `bootstrap` in the background

**Comment by AlbertShown** [comment]
> it will download library compiled for my specific OS

Well, this is an issue that needs to be fixed, for now, you can download binaries for all OS manually. At the end of the day, bootstrap script does nothing special, simply download .zip and decompress it.

**Comment by 7flash** [comment]
I am not sure what exactly solution looks like to make master branch work with Deno compile; will have to previous version for now where its simply using json imports

**Comment by hassandraga** [comment]
We are trying to write a downloader script that will be used in all wrappers. The bootstrap script just get updated. Now:

**Linux**
* To download one lib for current os: `sh bootstrap minimal`
* To download **all libs for all os**: `sh bootstrap`

**Windows**
* To download one lib for current os: `bootstrap.bat minimal`
* To download **all libs for all os**: `bootstrap.bat`

**Comment by 7flash** [comment]
Interesting, so it will compile all the libs into my binary executable? It seems to be consistent with previous behavior where imports happened at compile time as well, but would it be rather possible to fetch corresponding user OS library from github in runtime, I think that would be better to reduce my executable file size which already over 500mb

**Comment by hassandraga** [comment]
Yes, I was planning to do that as well. You can run `bootstrap minimal` to download one lib, or `bootstrap` to download all libs, or simply don't run anything, and deno-webui will run the bootstrap at runtime in the background to download the needed one lib.

I did not mention this because I make it working for Windows, but I still need to finish the Linux/macOS side:
https://github.com/webui-dev/deno-webui/blob/3b68252886f1cfd83f76c0d732020c15a8fcd491/deps.ts#L87

**Comment by 7flash** [comment]
Amazing, it has reduced my .exe bundle from 474mb to 81.1mb on windows

**Comment by 7flash** [comment]
I've been debugging the reasons it fails when running the binary now and found an issue with zip library:
https://github.com/gildas-lormeau/zip.js/issues/464

Seems like it relays on something that not available in deno compiled binary

Perhaps we can consider using another zip library?
